### ISSUE TITLE
Cannot connect to Server - No valid refresh token Available

### ISSUE THREAD

**Issue by titodamiani** [boecker-lab/sirius#71]
Hi,
I’m using SIRIUS (CLI version for Linux) on a computer cluster and I am having problem connecting to the server. I logged in before starting the computation (`sirius login -u <email> -p`) and run the following command:

`sirius -i <INPUT_DIRECTORY>  -o <OUTPUT_DIRECTORY> --maxmz=800 --cores 16 formula --profile=orbitrap --ppm-max=5.0 --ppm-max-ms2=7.5 zodiac fingerprint structure canopus`

The log file is here: https://drive.google.com/drive/folders/1eDXk-6Dh-mPzNUQmTgBvZJ4sZn0h-5UN?usp=sharing 
As you can see, it gets stuck connecting to the server (I guess when trying to run CSI:FingerID fingerprint prediction):
```
WARNING 00:49:43 - Error when try to connect to Server. Try again in 4.0s 
Cause: Not Logged in, No valid refresh token Available
```
This never happened with SIRIUS 4.9 so I guess it’s related to the new login system in SIRIUS 5? I looked at the profile information (`sirius login --show`) after logging in and it seems the token is valid for only 2 hours form the login. Is that the problem? Or there is something else I am missing? If so, is there a way to extend the validity of the token? Would that solve the issue?

Thanks!
Tito

**Comment by mfleisch** [comment]
Hey Tito,
thanks for reporting this problem. 
@lfnothias reported a similar problem but I was not able to reproduce it so far. But I think here was a proxy server involved.

> I looked at the profile information (sirius login --show) after logging in and it seems the token is valid for only 2 hours form the login.
This is only the lifetime of the `access_token` which is used to authorize at the CSI:FingerID application server. There is a second so called `refresh_token` which will automatically used to request a new access_token from the login server when needed.
This `refresh_token` is valid until it is actively invalidated. Long story short: The 2h hours are expected behaviour and should not affect the login state.

I few things we should check:
1. The tokens are stored in the .sirius-5.4 directory in the users home directory. Not sure how your compute cluster is organized, but  this directory needs to available on the node of the compute cluster where the job is actually running.
If some kind of container is involved you would have to mount the folder from the host system.
2. do you rely on some kind of proxy server configuration?

**Comment by titodamiani** [comment]
Hi Markus, 
I think in my case the problem is that the `.sirius-5.4` folder is available on the front-end computer, but not on the computing node of the cluster, as you mentioned. 

I tried to set the node’s `HOME` variable equal to the front-end’s home directory, but it didn’t find the `.sirius-5.4` folder. Therefore, I assume SIRIUS doesn’t use the `HOME` variable, does it? If so, which variable does it use?

For now, I managed to run it by copying the `.sirius-5.4` folder from the fron-end PC to the home directory of the computing node. Let me know if you need other details.

Thanks,
Tito

**Comment by mfleisch** [comment]
Hey Tito,
`HOME` should work.
* Have you checked whether the `HOME` variable is set in the same environment as SIRIUS is executed? If some kind of grid engine is involved this is a common issue.
* SIRIUS should print you a log message on startup telling you which directory it is using. It should look like this `Sirius Workspace Successfull initialized at: /home/fleisch/.sirius-5.4`. This might help debugging.

I will add a command line option that allows you to set the directory directly with the next update. I should have done that already ;-)

**Comment by titodamiani** [comment]
Hi Markus,
I did some tests and I think the reason is what you mentioned: SIRIUS is being executed in a different environment as the login token is stored.

First of all, I have to login every session on the front-end PC (`sirius login --email=myemail@gmail.com -p`) and the token is stored in `/storage/praha1/home/tdamiani/.sirius-5.4` (front-end PC)

However, when running SIRIUS directly on the node, the log file says:
`INFO    12:23:29 - Sirius Workspace Successfull initialized at: /storage/brno3-cerit/home/tdamiani/.sirius-5.4` (node PC)

So it seems that SIRIUS is being executed on the node, but the token in only available on the front-end (as I logged in form there). Could this be?

For now, I made it work by copying the `.sirius-5.4` folder form the front-end to the node’s home directory (by including `cp -r /storage/praha1/home/tdamiani/.sirius-5.4 ~/` in the script).

Thanks a lot!
Tito

P.S.: Feel free to close this issue:)

**Comment by titodamiani** [comment]
Hi,
I’m having again problems connecting to the server, but now due to a different reason apparently:

```
WARNING 15:20:57 - Error when try to connect to Server. Try again in 4.0s 
 Cause: Error when querying REST service. Bad Response Code: 404 | Message: Not Found | null | Content: <html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx</center>
</body>
</html>
```
 The full log file is [here](https://drive.google.com/drive/folders/1eDXk-6Dh-mPzNUQmTgBvZJ4sZn0h-5UN?usp=sharing). 
I manage to run a full SIRIUS workflow (see command above) about 7-10 days ago and nothing changed on my side, I think. What could it be the problem here?

Thanks a lot for the help!
Tito

**Comment by mfleisch** [comment]
Your still on the 5.4.1 pre-release?
It has been replaced by the official release 5.5.x
Upgrading to the latest version should solve the issue.

**Comment by titodamiani** [comment]
You're right! I was still running the 5.4.1 pre-release and upgrading to the latest version solved the issue. Thanks!
